tests/testsuite/login.rs

```diff
@@ -113,6 +113,7 @@ fn empty_login_token() {
         .with_stderr(
             "\
 [UPDATING] crates.io index
+please paste the token found on [..] below
```

Comment:
How is this test passing today?

Reply:
In the `read_token` function, if a token is provided by `stdin` or by the command line, then no prompt is made. 

The change causes `login_options.token` to be `None`, instead of a `Some("\n")`, so Cargo now prints the "please paste" message.
 
https://github.com/rust-lang/cargo/blob/852a31615d8ad66cf9768e16ef50119806629027/credential/cargo-credential/src/lib.rs#L281-L290

Reply:
Could you move that change into the commit that caused it so that each commit passes tests?

Reply:
Fixed. I should have amended that instead of adding a commit.